src/main/java/org/broadinstitute/hellbender/utils/variant/GATKVariantContextUtils.java

```diff
@@ -340,22 +341,23 @@ public static void makeGenotypeCall(final int ploidy,
         }
     }
 
-    private static double getGQLog10FromPosteriors(final int bestGenotypeIndex, final double[] /**/log10Posteriors) {
+    @VisibleForTesting
+    static double getGQLog10FromPosteriors(final int bestGenotypeIndex, final double[] /**/log10Posteriors) {
         if (bestGenotypeIndex < 0) {
             return CommonInfo.NO_LOG10_PERROR;
         } else {
             switch (log10Posteriors.length) {
                 case 0:
                 case 1: return CommonInfo.NO_LOG10_PERROR;
                 case 2: return bestGenotypeIndex == 0 ? log10Posteriors[1] : log10Posteriors[0];
-                case 3: return Math.min(0, MathUtils.log10SumLog10(
+                case 3: return Math.min(0.0, MathUtils.log10SumLog10(
                                  log10Posteriors[ bestGenotypeIndex == 0 ? 2 : bestGenotypeIndex - 1],
                                  log10Posteriors[ bestGenotypeIndex == 2 ? 0 : bestGenotypeIndex + 1]));
                 default:
                     if (bestGenotypeIndex == 0) {
-                        return MathUtils.log10SumLog10(log10Posteriors, 1, log10Posteriors.length);
+                        return Math.min(0.0, MathUtils.log10SumLog10(log10Posteriors, 1, log10Posteriors.length));
```

Comment:
Another issue for @vruano : If we have to throw a min on this, then the normalization isn't happening in the right place.  According to my calculations (and as seen in James's unit test) the common case where this comes into play is where there are more than two equally likely genotypes, so the log10 probabilities get normalized to [0, 0, 0, -something, more negative somethings], the best is a zero and then there are two not bests that are zero and the log10SumLog10(0,0) makes a log10(2) ~ 0.3 > 0 which leads to a negative GQ in Phred-space.  Elsewhere we use the MathUtils.secondSmallestMinusSmallest, which doesn't have this problem.  I don't think the log10SumLog10 is the right thing to do.

Reply:
Based on the definition of GQ in the VCF we are supposed to do that.

Reply:
Given that we output integer GQ values (as in the spec), the "smallest minus second smallest" will be the same as 1-Prob(GT)/Z unless the PLs are [0,0,0] right?